src/components/CmsZone.vue

```diff
@@ -275,40 +275,48 @@ export default class CmsZone extends Vue {
     cmsClient.trackZone({ content, zoneId: this.zoneId });
   }
 
-  private trackScrollable(contents: Content[], scrollable: Element): void {
-    if (this.zoneType === 'carousel') {
-      let timeout: number;
-      const listener = (): void => {
-        window.clearTimeout(timeout);
-        timeout = window.setTimeout((): void => {
-          const contentElm = this.$el.querySelector('.slick-current');
-          if (contentElm && this.isContentVisible(contentElm, scrollable, percentVisible)) {
-            this.trackIndex(0);
-            scrollable.removeEventListener('scroll', listener);
-          }
-        }, durationVisibleToBeTrackedMs);
+  private addScrollListener(callback: () => void | Promise<void>, delay: number = 0): () => void {
+    if (delay) {
+      const rawCallback = callback;
+      let timer: number;
+      callback = (): void => {
+        window.clearTimeout(timer);
+        timer = window.setTimeout(rawCallback, delay);
       };
-      scrollable.addEventListener('scroll', listener);
-      this.scrollableListeners.push(listener);
-      Vue.nextTick(listener);
+    }
+
+    const listener = debounce(callback, 100, { leading: true, trailing: true });
```

Comment:
should this be `debounce` or `throttle`? I think we still want the listener to be called as the user scrolls

Reply:
didn't know about throttle, will change it.